pkg/util/trivy/cache.go

```diff
@@ -9,16 +9,241 @@
 		cacheDir = utils.DefaultCacheDir()
 	}
 
 	return cache.NewFSCache(cacheDir)
 }
+
+// BadgerCache implements the Trivy Cache interface. It's provided as an
+// alternative to BoltDB, the cache used by default in Trivy. The main advantage
+// of Badger is that it allows to specify a TTL for the objects stored in the
+// cache.
+type BadgerCache struct {
+	db                      *badger.DB
+	ttl                     time.Duration
+	garbageCollectionTicker *time.Ticker
+}
+
+// This is needed so that Badger logs using the same format as the Agent
+type badgerLogger struct{}
+
+func (l badgerLogger) Errorf(format string, params ...interface{}) {
+	log.Errorf(format, params...)
+}
+
+func (l badgerLogger) Warningf(format string, params ...interface{}) {
+	log.Warnf(format, params...)
+}
+
+func (l badgerLogger) Infof(format string, params ...interface{}) {
+	log.Infof(format, params...)
+}
+
+func (l badgerLogger) Debugf(format string, params ...interface{}) {
+	log.Debugf(format, params...)
+}
+
+func NewBadgerCache(cacheDir string, ttl time.Duration) (*BadgerCache, error) {
+	if cacheDir == "" {
+		cacheDir = filepath.Join(os.TempDir(), "datadog-agent-sbom-cache")
+	}
+
+	db, err := badger.Open(badger.DefaultOptions(cacheDir).WithLogger(badgerLogger{}))
+	if err != nil {
+		return nil, fmt.Errorf("error creating Badger cache: %w", err)
+	}
+
+	// Run garbage collection periodically. It's needed because Badger relies on
+	// the client to run the garbage collection
+	// (https://dgraph.io/docs/badger/get-started/#garbage-collection).
+	gcTicker := time.NewTicker(garbageCollectionFrequency)
+	go func() {
+		for range gcTicker.C {
+			if err := db.RunValueLogGC(garbageCollectionDiscardRatio); err != nil && err != badger.ErrNoRewrite {
+				// ErrNoRewrite is returned when the GC runs fine but doesn't
+				// result any cleanup. That's why we don't log anything in that
+				// case.
+				log.Warnf("error while running garbage collection in Badger: %s", err)
+			}
+		}
+	}()
+
+	return &BadgerCache{db: db, ttl: ttl, garbageCollectionTicker: gcTicker}, nil
+}
+
+func (cache *BadgerCache) MissingBlobs(artifactID string, blobIDs []string) (bool, []string, error) {
+	var missingArtifact bool
+	var missingBlobIDs []string
+
+	err := cache.db.View(func(txn *badger.Txn) error {
+		if _, err := txn.Get([]byte(artifactID)); err != nil {
+			if err == badger.ErrKeyNotFound {
+				missingArtifact = true
+			} else {
+				return err
+			}
+		}
+
+		for _, blobID := range blobIDs {
+			if _, err := txn.Get([]byte(blobID)); err != nil {
+				if err == badger.ErrKeyNotFound {
+					missingBlobIDs = append(missingBlobIDs, blobID)
+				} else {
+					return err
+				}
+			}
+		}
+
+		return nil
+	})
+	if err != nil {
+		return false, nil, fmt.Errorf("error getting missing blobs from Badger cache: %w", err)
+	}
+
+	return missingArtifact, missingBlobIDs, nil
+}
+
+func (cache *BadgerCache) PutArtifact(artifactID string, artifactInfo types.ArtifactInfo) error {
```

Comment:
`PutArtifact` and `PutBlob` seems to contain duplicated code.
May I suggest a way to factorize those functions in #15948 ?